src/report/generateFileCoverageHtml.ts

```diff
@@ -58,28 +60,99 @@ const generateFileCoverageHtml = ({
 								jsonFinal,
 								commitSHA,
 								workspacePath,
+								comparisonDecimalPlaces,
 							),
 						)
 						.join("")}
 				`;
 	}
 
-	if (fileCoverageMode === FileCoverageMode.All && unchangedFiles.length > 0) {
-		reportData += `
-						${formatGroupLine("Unchanged Files")}
-						${unchangedFiles
-							.map((filePath) =>
-								generateRow(
-									filePath,
-									jsonSummary,
-									undefined,
-									jsonFinal,
-									commitSHA,
-									workspacePath,
-								),
-							)
-							.join("")}
+	if (
+		(fileCoverageMode === FileCoverageMode.ChangesAffected ||
+			fileCoverageMode === FileCoverageMode.All) &&
+		unchangedFiles.length > 0 &&
+		jsonSummaryCompare
+	) {
+		// Split unchanged files into affected and unaffected
+		const [affectedFiles, unaffectedFiles] = splitFilesByCoverageChange(
+			unchangedFiles,
+			jsonSummary,
+			jsonSummaryCompare,
+		);
+
+		if (affectedFiles.length > 0) {
+			reportData += `
+					${formatGroupLine("Affected Files")}
+					${affectedFiles
+						.map((filePath) =>
+							generateRow(
+								filePath,
+								jsonSummary,
+								jsonSummaryCompare,
+								jsonFinal,
+								commitSHA,
+								workspacePath,
+								comparisonDecimalPlaces,
+							),
+						)
+						.join("")}
+				`;
+		}
+
+		// In All mode, also show unaffected files
+		if (
+			fileCoverageMode === FileCoverageMode.All &&
+			unaffectedFiles.length > 0
+		) {
+			reportData += `
+					${formatGroupLine("Unaffected Files")}
+					${unaffectedFiles
+						.map((filePath) =>
+							generateRow(
+								filePath,
+								jsonSummary,
+								jsonSummaryCompare,
+								jsonFinal,
+								commitSHA,
+								workspacePath,
+								comparisonDecimalPlaces,
+							),
+						)
+						.join("")}
 				`;
+		}
+	} else if (
+		fileCoverageMode === FileCoverageMode.ChangesAffected &&
+		unchangedFiles.length > 0 &&
+		!jsonSummaryCompare
+	) {
+		// Add a note that comparison data is needed for affected files
+		reportData += `
+			<tr>
+				<td colspan="6"><em>Note: Comparison data is required to show affected files. Provide <code>json-summary-compare-path</code> to enable this feature.</em></td>
+			</tr>
+		`;
+	} else if (
+		fileCoverageMode === FileCoverageMode.All &&
+		unchangedFiles.length > 0
+	) {
+		// Fallback if no comparison data: show all unchanged files together
+		reportData += `
+					${formatGroupLine("Unchanged Files")}
+					${unchangedFiles
```

Comment:
For unchanged files, we don't need a comparison (it's going to be equal all the time anyway)